applications/serial_lte_modem/src/main.c

```diff
@@ -411,10 +401,13 @@ int main(void)
 	if (slm_settings_init() != 0) {
 		LOG_WRN("Failed to init slm settings");
 	}
+
+	err = nrf_modem_lib_init(NORMAL_MODE);
```

Comment:
This seems to be not good for FMFU?

Reply:
For FMFU, the modem library has to be shutdown and reinitialized in bootloader mode. However, unless I have overseen something, it will behave the same as before. When `CONFIG_NRF_MODEM_LIB_SYS_INIT` is enabled, the modem library is initialized in normal mode.

Reply:
The code to support FMFU is not in Main.
We can modify later.

Reply:
nrf_modem_lib_init() @return int Zero on success, non-zero otherwise.
The `err` is not the one to be handled be handle_nrf_modem_lib_init_ret(err);
Instead the main() should terminate here and do a reboot right away.

Reply:
I have rewritten this locally to 
```
err = nrf_modem_lib_init(NORMAL_MODE);
	if (err && (fota_stage == FOTA_STAGE_INIT)) {
		LOG_ERR("Modem library init failed, err: %d", err);
		return err;
	}
```
When performing DFU, `err` will contain the DFU result unless `err` is negative, which is a modem library error. Negative `err` is handled by the default case in `handle_nrf_modem_lib_init_ret()`.

Reply:
https://github.com/nrfconnect/sdk-nrfxlib/blob/main/nrf_modem/include/nrf_modem.h#L218
Not seen from here

Reply:
Yes, I've created a PR to update this: [sdk-nrfxlib#938](https://github.com/nrfconnect/sdk-nrfxlib/pull/938)